### PR TITLE
feat: modify api syntax to be compatible with older versions of Node

### PR DESCRIPTION
**Pull request by jeffreyzhang23**
### 🔗 Linked issue

[#119](https://github.com/unjs/magicast/issues/119)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Modify some code to be compatible with older Node version, such as "String.prototype.at".
Resolves [#119](https://github.com/unjs/magicast/issues/119)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: scripts/vendor.ts
Comment (jeffreyzhang23) [inline scripts/vendor.ts:1]: My changes are on lines 19 and 77.

**Comment by jeffreyzhang23** [comment]
Looking forward to your reply. @antfu

**Comment by antfu** [comment]
Node 14 reached EOL over a year ago. We don't have a plan to support that. Please consider upgrade your node instead. Thanks

Btw, just a reminder, when you sending PR the next time, it's better to disable your formatter. Regardless the implementation, PR like this that changes all formatting would not get accpeted.

**Comment by jeffreyzhang23** [comment]
> Node 14 reached EOL over a year ago. We don't have a plan to support that. Please consider upgrade your node instead. Thanks
> 
> Btw, just a reminder, when you sending PR the next time, it's better to disable your formatter. Regardless the implementation, PR like this that changes all formatting would not get accpeted.

Thank you for your advice! 
In fact, not only Node 14, versions below Node 16.6.0 cannot use magicast ([String.prototype.at minimum support 16.6.0](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/at)). Moreover, this syntax is replaceable, allowing other versions to also benefit from such an excellent tool. @antfu

**Comment by jeffreyzhang23** [comment]
> > Node 14 reached EOL over a year ago. We don't have a plan to support that. Please consider upgrade your node instead. Thanks
> > Btw, just a reminder, when you sending PR the next time, it's better to disable your formatter. Regardless the implementation, PR like this that changes all formatting would not get accpeted.
> 
> Thank you for your advice! In fact, not only Node 14, versions below Node 16.6.0 cannot use magicast ([String.prototype.at minimum support 16.6.0](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/at)). Moreover, this syntax is replaceable, allowing other versions to also benefit from such an excellent tool. @antfu

If possible, I will submit a more standardized PR. Thank you!

**Comment by antfu** [comment]
Node 16 is also EOL... The cases is that we can't keep lowering the syntax just because someone using the old Node version.

**Comment by jeffreyzhang23** [comment]
> Node 16 is also EOL... The cases is that we can't keep lowering the syntax just because someone using the old Node version.

I got it. Thanks for the reply ❤️.

**Pull request closed without merging**